ts-client/src/dlmm/types/index.ts

```diff
@@ -188,6 +189,57 @@ export interface BinLiquidity {
   pricePerToken: string;
 }
 
+export module BinLiquidity {
+  export function fromBin(
+    bin: Bin,
+    binId: number,
+    binStep: number,
+    baseTokenDecimal: number,
+    quoteTokenDecimal: number,
+    version?: number,
+  ): BinLiquidity {
+    const pricePerLamport = getPriceOfBinByBinId(
+      binId,
+      binStep
+    ).toString();
+    return {
+      binId,
+      xAmount: bin.amountX,
+      yAmount: bin.amountY,
+      supply: bin.liquiditySupply,
+      price: pricePerLamport,
+      version: version ?? 1,
+      pricePerToken: new Decimal(pricePerLamport)
+        .mul(new Decimal(10 ** (baseTokenDecimal - quoteTokenDecimal)))
+        .toString(),
+    };
+  }
+
+  export function empty(
+    binId: number,
+    binStep: number,
+    baseTokenDecimal: number,
+    quoteTokenDecimal: number,
+    version?: number,
+  ): BinLiquidity {
+    const pricePerLamport = getPriceOfBinByBinId(
+      binId,
+      binStep
+    ).toString();
+    return {
+      binId,
+      xAmount: new BN(0),
+      yAmount: new BN(0),
+      supply: new BN(0),
+      price: pricePerLamport,
+      version: version ?? 1,
```

Comment:
ditto ^

Reply:
> ditto ^

I see if no bin array found v1 is used:
https://github.com/MeteoraAg/dlmm-sdk/blob/1d4c4b9fa3bd60770dfdbd481d664ae52587d943/ts-client/src/dlmm/index.ts#L5714
I follow the same rule. I assume if there's any bin array found I should pass it, right?